src/cpp/utils.cpp

```diff
@@ -0,0 +1,33 @@
+#include "utils.h"
+
+#include <fstream>
+#include <iomanip>
+
+namespace effil {
+
+Logger getLogger() {
+    static std::mutex lock;
+    static const char* logFile = getenv("EFFIL_LOG");
+
+    if (logFile == nullptr) {
+        static std::ostream devNull(0);
+        return Logger(devNull, lock);
+    }
+    else if (strcmp(logFile, "term") == 0) {
+        return Logger(std::cout, lock);
+    }
+    else {
+        static std::ofstream fileStream(logFile);
```

Comment:
Static vars are very-very bad. They will lead to crashes on deinitialization eventually. 
You can write singleton like this.
```c++
Logger& getLogger() {
    static auto loggerPtr = new Logger();
    return *loggerPtr.
}
```
It is safe but not frees the memory. 
Or use something like this singleton https://github.com/boostorg/serialization/blob/develop/include/boost/serialization/singleton.hpp